pkg/local_object_storage/blobstor/common/storage.go

```diff
@@ -23,3 +27,53 @@ type Storage interface {
 	Delete(DeletePrm) (DeleteRes, error)
 	Iterate(IteratePrm) (IterateRes, error)
 }
+
+// Copy copies all objects from source Storage into the destination one. If any
+// object cannot be stored, Copy immediately fails.
+func Copy(from, to Storage) error {
+	err := from.Open(true)
+	if err != nil {
+		return fmt.Errorf("open source sub-storage: %w", err)
+	}
+
+	defer func() { _ = from.Close() }()
+
+	err = to.Open(false)
+	if err != nil {
+		return fmt.Errorf("open destination sub-storage: %w", err)
+	}
+
+	defer func() { _ = to.Close() }()
+
+	err = to.Init()
```

Comment:
why don't you init `from`?

Reply:
usually not needed for read-only ops, but I agree it won't hurt